src/index.js

```diff
@@ -53,9 +53,39 @@ function isReactClass(superClass, scope, globalOptions) {
   return answer
 }
 
+function areSetsEqual(set1, set2) {
+  if (set1.size !== set2.size) {
```

Comment:
even tho it doesn't matter for this use case, i'd probably add an additional bailout; `if (set1 === set2) { return true; }` :-)